### PR TITLE
SDF vs YANG

### PR DESCRIPTION
**Pull request by mcr**

### CODE REVIEW

**Review by nwidell — APPROVED** [review]
Looks good to me (but I'm not a YANG person).

**Review by WAvdBeek — APPROVED** [review]

File: charter.md [hunk 1]
```diff
@@ -36,6 +36,17 @@
   ecosystems in a common way; they look towards the IETF for
   developing SDF 1.0 further into a high-quality specification.
 
+  The IETF also defines the YANG data modelling language (RFC 7950).
+  YANG has a strong focus on modelling the management interface for network devices,
+  using a fairly small set of network management protocols,
+  whereas SDF is designed to span (and transitively unify) many existing interaction models already present.
+  While YANG can be serialized to a variety of wire formats (XML, JSON, CBOR, ...),
+  the serialization is driven directly by the structure of the YANG.
```
Comment (cabo) [inline charter.md]: ; there is no separate "protocol binding" step as in the use of SDF.

File: charter.md [hunk 2]
```diff
@@ -36,6 +36,17 @@
...
+
+  YANG also has an opinionated way in which the RPC is done.
+  This is necessary in order to get the kind of cross-vendor interoperability needed for efforts in the Software Defined Networking.
+  On the other hand, SDF does not deal directly with serialization, dealing only in semantics of content, and leaving the existing IoT SDOs to do their own serialization.
```
Comment (cabo) [inline charter.md]: I think I can live with s/semantics/structure and semantics/, as was suggested.
Also, s/content/data being interchanged/

File: charter.md (hunk 1 above)
Comment (mcr) [reply charter.md]: I don't know how to process this. Is it a suggestion?

Comment (mcr) [inline charter.md]:
```suggestion
  the serialization is driven directly by the structure of the YANG; there is no separate "protocol binding" step as in the use of SDF.
```

Comment (mcr) [reply charter.md]: Is this what you had in mind?

File: charter.md (hunk 2 above)
Comment (mcr) [inline charter.md]:
```suggestion
  On the other hand, SDF does not deal directly with serialization, dealing only in structure and semantics of data being interchanged, and leaving the existing IoT SDOs to do their own serialization.
```

File: charter.md (hunk 1 above)
Comment (mcr) [inline charter.md:44]:
```suggestion
    While data associated with a YANG schema can be serialized to a variety
   of wire formats (e.g., XML, JSON, CBOR, and others), the serialization is
   driven directly by the structure of the data models defined in YANG; there is no separate "protocol binding" step as in the use of SDF.
```

File: charter.md (hunk 2 above)
Comment (mcr) [inline charter.md]:
```suggestion
   Conversely, SDF does not deal directly with serialization at all,
   modelling only the structure and semantics of the data being interchanged, hence leaving data
   serialization (and RPC semantics) to other standards, most likely defined
   by existing IoT SDOs.

```

File: charter.md [hunk 3]
```diff
@@ -36,6 +36,17 @@
   ecosystems in a common way; they look towards the IETF for
   developing SDF 1.0 further into a high-quality specification.
 
+  The IETF also defines the YANG data modelling language (RFC 7950).
+  YANG has a strong focus on modelling the management interface for network devices,
+  using a fairly small set of network management protocols,
+  whereas SDF is designed to span (and transitively unify) many existing interaction models already present.
+  While YANG can be serialized to a variety of wire formats (XML, JSON, CBOR, ...),
+  the serialization is driven directly by the structure of the YANG; there is no separate "protocol binding" step as in the use of SDF.
```
Comment (mcr) [inline charter.md]:
```suggestion
    While data associated with a YANG schema can be serialized to a variety
   of wire formats (e.g., XML, JSON, CBOR, and others), the serialization is
   driven directly by the structure of the data models defined in YANG; there is no separate "protocol binding" step as in the use of SDF.
```

File: charter.md [hunk 4]
```diff
@@ -36,6 +36,17 @@
...
+
+  YANG also has an opinionated way in which the RPC is done.
+  This is necessary in order to get the kind of cross-vendor interoperability needed for efforts in the Software Defined Networking.
+  On the other hand, SDF does not deal directly with serialization, dealing only in structure and semantics of data being interchanged, and leaving the existing IoT SDOs to do their own serialization.
+
```
Comment (mcr) [inline charter.md]:
```suggestion
   Conversely, SDF does not deal directly with serialization at all,
   modelling only the structure and semantics of the data being interchanged, hence leaving data
   serialization (and RPC semantics) to other standards, most likely defined
   by existing IoT SDOs.

```

File: charter.md (hunk 1 above)
Comment (cabo) [reply charter.md]: Well, you just add the text after "the structure of the YANG" (or what you now made of that).

File: charter.md (hunk 2 above)
Comment (cabo) [reply charter.md]: I cannot apply this, but it seems to be a good change.  Wow, github can be very confusing...

File: charter.md [hunk 5]
```diff
@@ -36,6 +36,21 @@
...
+  The IETF also defines the YANG data modelling language (RFC 7950).
+  YANG has a strong focus on modelling the management interface for network devices,
+  using a fairly small set of network management protocols,
+  whereas SDF is designed to span (and transitively unify) many existing interaction models already present.
+    While data associated with a YANG schema can be serialized to a variety
+   of wire formats (e.g., XML, JSON, CBOR, and others), the serialization is
+   driven directly by the structure of the data models defined in YANG; there is no separate "protocol binding" step as in the use of SDF.
+
+  YANG also has an opinionated way in which the RPC is done.
+  This is necessary in order to get the kind of cross-vendor interoperability needed for efforts in the Software Defined Networking.
```
Comment (mcr) [inline charter.md]:
```suggestion
```

**Pull request merged**